sklearn/metrics/tests/test_pairwise.py

```diff
@@ -195,6 +197,24 @@ def test_pairwise_distances(global_dtype):
     with pytest.raises(ValueError):
         pairwise_distances(X, Y, metric="blah")
 
+    # Test using custom metrics, serial and parallel
+    def customMetric(x0, x1):
```

Comment:
This test does not catch issues that happen when there is more than one feature. I recommend adding this new test that checks that the distance matrix is consistent betwen the three methods of computation:

```python
def test_pairwise_custom_metric_multithread_and_loky():
    # Test using custom metrics, serial and parallel
    def custom_metric(x0, x1):
        return np.abs(x0 - x1).sum()

    rng = np.random.RandomState(0)
    X = rng.randn(10, 4)

    n_jobs = 2
    # default parallel approach, preferring multithread
    dist_multithread = pairwise_distances(X, metric=custom_metric, n_jobs=n_jobs)
    # parallel with loky
    with parallel_backend("loky", n_jobs=n_jobs):
        dist_loky = pairwise_distances(X, metric=custom_metric, n_jobs=n_jobs)
    dist_serial = pairwise_distances(X, metric=custom_metric)
    assert_allclose(dist_multithread, dist_serial)
    assert_allclose(dist_loky, dist_serial)
```

Reply:
Thanks Thomas, for the useful suggestions, I'll do as you recommend.
Anyway, IMHO, this feature can be useful, as the only way I know to release the GIL would be writing a native-code extension, e.g. in c, depriving many pure-python users from the benefits of parallelization for this functionality.